pkg/operator/ceph/object/spec.go

```diff
@@ -314,42 +328,34 @@ func (c *clusterConfig) makeDaemonContainer(rgwConfig *rgwConfig) v1.Container {
 	}
 	kmsEnabled, err := c.CheckRGWKMS()
 	if err != nil {
-		logger.Errorf("failed to enable KMS. %v", err)
+		logger.Errorf("failed to enable SSE-KMS. %v", err)
 		return v1.Container{}
 	}
 	if kmsEnabled {
-		container.Args = append(container.Args,
-			cephconfig.NewFlag("rgw crypt s3 kms backend",
-				c.store.Spec.Security.KeyManagementService.ConnectionDetails[kms.Provider]),
-			cephconfig.NewFlag("rgw crypt vault addr",
-				c.store.Spec.Security.KeyManagementService.ConnectionDetails[api.EnvVaultAddress]),
-		)
+		container.Args = append(container.Args, c.sseKMSDefaultOptions()...)
 		if c.store.Spec.Security.KeyManagementService.IsTokenAuthEnabled() {
-			container.Args = append(container.Args,
-				cephconfig.NewFlag("rgw crypt vault auth", kms.KMSTokenSecretNameKey),
-				cephconfig.NewFlag("rgw crypt vault token file",
-					path.Join(rgwVaultDirName, kms.VaultFileName)),
-				cephconfig.NewFlag("rgw crypt vault prefix", c.vaultPrefixRGW()),
-				cephconfig.NewFlag("rgw crypt vault secret engine",
-					c.store.Spec.Security.KeyManagementService.ConnectionDetails[kms.VaultSecretEngineKey]),
-			)
+			container.Args = append(container.Args, c.sseKMSVaultTokenOptions()...)
 		}
 		if c.store.Spec.Security.KeyManagementService.IsTLSEnabled() &&
-			c.clusterInfo.CephVersion.IsAtLeast(cephver.CephVersion{Major: 16, Minor: 2, Extra: 6}) {
-			container.Args = append(container.Args,
-				cephconfig.NewFlag("rgw crypt vault verify ssl", "true"))
-			if kms.GetParam(c.store.Spec.Security.KeyManagementService.ConnectionDetails, api.EnvVaultClientCert) != "" {
-				container.Args = append(container.Args,
-					cephconfig.NewFlag("rgw crypt vault ssl clientcert", path.Join(rgwVaultDirName, kms.VaultCertFileName)))
-			}
-			if kms.GetParam(c.store.Spec.Security.KeyManagementService.ConnectionDetails, api.EnvVaultClientKey) != "" {
-				container.Args = append(container.Args,
-					cephconfig.NewFlag("rgw crypt vault ssl clientkey", path.Join(rgwVaultDirName, kms.VaultKeyFileName)))
-			}
-			if kms.GetParam(c.store.Spec.Security.KeyManagementService.ConnectionDetails, api.EnvVaultCACert) != "" {
-				container.Args = append(container.Args,
-					cephconfig.NewFlag("rgw crypt vault ssl cacert", path.Join(rgwVaultDirName, kms.VaultCAFileName)))
-			}
+			c.clusterInfo.CephVersion.IsAtLeast(cephVersionMinRGWSSETLS) {
+			container.Args = append(container.Args, c.sseKMSVaultTLSOptions()...)
+		}
+		vaultVolMount := v1.VolumeMount{Name: rgwVaultVolumeName, MountPath: rgwVaultDirName}
+		container.VolumeMounts = append(container.VolumeMounts, vaultVolMount)
+	}
+
+	s3Enabled, err := c.CheckRGWS3()
```

Comment:
Anything that could cause an error should be called before or after this outer method. The only purpose of this method should be container spec generation. This will allow the reconcile to also fail instead of swallowing the error. So `s3Enabled` likely should be a parameter to this method.